ydb/core/tx/tx_proxy/proxy_ut.cpp

```diff
@@ -1062,7 +1062,40 @@ Y_UNIT_TEST_SUITE(TModifyUserTest) {
             TString ldapUserToken = CheckLogin(client, env.GetRuntime(), "user@ldap", "ldapuser");
 
             // ldap user cannot self change password
-            UNIT_ASSERT_VALUES_EQUAL(client.ModifyUser("/dc-1", "user@ldap", "ldapuserpass", ldapUserToken), NMsgBusProxy::MSTATUS_ERROR);
+            UNIT_ASSERT_VALUES_EQUAL(client.ModifyUser("/dc-1", { .User = "user@ldap", .Password = "ldapuserpass"}, ldapUserToken), NMsgBusProxy::MSTATUS_ERROR);
+        }
+    }
+
+    Y_UNIT_TEST(ModifyUserIsEnabled) {
+        TTestEnv env;
+
+        {
+            auto subdomain = GetSubDomainDeclareSetting("USER_0", env.GetPools());
+            UNIT_ASSERT_VALUES_EQUAL(NMsgBusProxy::MSTATUS_OK, env.GetClient().CreateSubdomain("/dc-1", subdomain));
+        }
+
+        auto& client = env.GetClient();
+        TString user1Token;
+        {
+            client.CreateUser("/dc-1", { .User = "user1", .Password = "pass1"}, "root@builtin");
+            user1Token = CheckLogin(client, env.GetRuntime(), "user1", "pass1");
+        }
+
+        TString user2Token;
+        {
+            client.CreateUser("/dc-1", { .User = "user2", .Password = "pass2"}, "root@builtin");
+            user2Token = CheckLogin(client, env.GetRuntime(), "user2", "pass2");
+        }
+
+        {
+            // user2 cannot change isEnabled for user1. user2 has not ydb.granular.alter_schema permission
+            UNIT_ASSERT_VALUES_EQUAL(client.ModifyUser("/dc-1", { .User = "user1", .CanLogin = Tests::TClient::ETypeOfLogin::NoLogin}, user2Token), NMsgBusProxy::MSTATUS_ERROR);
+
+            // user cannot change isEnabled for self
+            UNIT_ASSERT_VALUES_EQUAL(client.ModifyUser("/dc-1", { .User = "user2", .CanLogin = Tests::TClient::ETypeOfLogin::NoLogin}, user2Token), NMsgBusProxy::MSTATUS_ERROR);
```

Comment:
Can user change `isEnabled` for self if he will has permission `ydb.granular.alter_schema`?